README.md

```diff
@@ -25,32 +24,52 @@ extensions:
 codeUrl: https://github.com/microsoftgraph/dotnet-aad-query-sample
 zipUrl: https://github.com/microsoftgraph/dotnet-aad-query-sample/archive/master.zip
 ---
+# Explore advanced Microsoft Graph Query Capabilities for Directory Objects with .NET SDK
 
-# Advanced Query Capabilities for Directory Objects with .NET SDK
+- [Explore advanced Microsoft Graph Query Capabilities for Directory Objects with .NET SDK](#explore-advanced-microsoft-graph-query-capabilities-for-directory-objects-with-net-sdk)
+  - [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Setup](#setup)
+    - [Step 1: Register your application](#step-1-register-your-application)
+    - [Step 2: Set the permissions](#step-2-set-the-permissions)
+    - [Step 3: Configure the ClientId using the Secret Manager](#step-3-configure-the-clientid-using-the-secret-manager)
+  - [Build & Run](#build--run)
+    - [Using the app](#using-the-app)
+  - [Code Architecture](#code-architecture)
+
+## Overview
 
 This sample helps you explore the Microsoft Graph's [new query capabilities](https://aka.ms/BlogPostMezzoGA) of the identity APIs using the [Microsoft Graph SDK](https://github.com/microsoftgraph/msgraph-sdk-dotnet) to query Azure AD.
 The main code is in [GraphDataService.cs](MSGraphSamples.WPF/Services/GraphDataService.cs) file, where every request pass trough `AddAdvancedOptions` function adding the required `$count=true` QueryString parameter and `ConsistencyLevel=eventual` header.
 
-## Getting Started
+## Prerequisites
+
+- Visual Studio
+- An Internet connection
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/documentation/articles/active-directory-howto-tenant/)
+- A user account in your Azure AD tenant. This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.
+
+## Setup
 
-### 1\. Register your application
+### Step 1: Register your application
 
 Use the [Microsoft Application Registration Portal](https://aka.ms/appregistrations) to register your application with the Microsoft Graph APIs.
 ![Application Registration](docs/register_app.png)
```

Comment:
Is admin consent needed to be granted on **Permissions** blade? Readme screenshot suggests it is, but it does not seem necessary

![admin_consent_needed](https://user-images.githubusercontent.com/13197548/101834495-4ee29900-3aef-11eb-99d4-2b8a802da680.png)

Reply:
It's a bit ambiguous because of this note that @kalyankrishna1 mentioned during our meeting:
![image](https://user-images.githubusercontent.com/2388430/101847679-9fb2bb80-3b08-11eb-91d0-15c122d091aa.png)

I would keep it as is since if they are registering the app in another tenant, then it needs admin consent.